### PR TITLE
test: Add unit tests for is_closing_session()

### PR DESCRIPTION
**Pull request by bdrung**
Checking for root processes does not belong into `is_closing_session`.

Add unit tests for `is_closing_session()` to increase code coverage to ease future refactoring.

### CODE REVIEW

**Comment by schopin-pro** [comment]
I'm guessing it conflicts with the `woke` renaming work, given the negative diff contains "sanity"?

File: data/apport [hunk 1]
```diff
@@ -1031,7 +1027,8 @@ def process_crash(options: argparse.Namespace) -> int:
     # write out the user coredumps
     recover_privileges()
 
-    if is_closing_session():
+    # Do not check closing session for root processes
+    if crash_uid != 0 and crash_gid != 0 and is_closing_session():
```
Comment (schopin-pro) [inline data/apport:1031]: question: why do we want to special-case root here?

I mean I have a vague idea why, but that's somewhat fuzzy.

**Review by schopin-pro — APPROVED** [review]
LGTM, I only have a question that's only tangentially related to the PR's purpose.

File: data/apport (hunk 1 above)
Comment (bdrung) [reply data/apport:1031]: Isn't "Checking for root processes does not belong into `is_closing_session`." explanition enough?

Comment (schopin-pro) [reply data/apport:1034]: I mean I understand why you moved it out, it makes sense, I'm more wondering why we had the check in the first place. As I said, only tangentially related ;).

I'll merge anyway.

Comment (bdrung) [reply data/apport:1034]: Let me search in the git history and improve the comment.

Comment (bdrung) [reply data/apport:1034]: It was introduced in commit e1c90227666d9b0f95c3cd6229df1f9097974f4f to fix CVE-2022-28655. It still does not explain the full reason.

**Pull request merged**